src/Tasks/Common/ConflictResolution/ConflictItem.cs

```diff
@@ -158,11 +158,33 @@ public string PackageId
             {
                 if (_packageId == null)
                 {
-                    _packageId = OriginalItem?.GetMetadata(MetadataNames.NuGetPackageId) ?? String.Empty;
+                    _packageId = OriginalItem?.GetMetadata(MetadataNames.NuGetPackageId);
+
+                    if (string.IsNullOrEmpty(_packageId))
+                    {
+                        _packageId = OriginalItem?.GetMetadata(MetadataKeys.PackageName) ?? string.Empty;
+                    }
 
                     if (_packageId.Length == 0)
                     {
-                        _packageId = NuGetUtils.GetPackageIdFromSourcePath(SourcePath) ?? String.Empty;
+                        //  We want to move away from using the heuristic of walking up the folder tree until
+                        //  we find a .nuspec in order to determine the package ID of a file.  However, we
+                        //  don't want to accidentally stop having a package ID for a file that the heuristic
+                        //  would have found the package ID for.  So to catch those cases, we throw an
+                        //  exception if the heuristic finds a package ID but we don't have the package ID
+                        //  from any other source
+                        string packageIdFromPath = NuGetUtils.GetPackageIdFromSourcePath(SourcePath);
+                        if (!string.IsNullOrEmpty(packageIdFromPath))
+                        {
+                            string path = OriginalItem?.ItemSpec;
+                            if (string.IsNullOrEmpty(path))
+                            {
+                                path = SourcePath;
+                            }
+                            throw new InvalidOperationException("NuGetPackageId metadata not set on " + path);
```

Comment:
Judging from the description, this is an internal diagnostic meant to weed out instances where our assets aren't emitting with the correct metadata and therefore this doesn't need to be loc'd, correct?

Reply:
Yes, the idea is eventually we just remove this whole section of logic.

Reply:
What happens for projects using packages.config?  Won't they always depend on this path?

Reply:
.NET Core projects never use packages.config.  Under what circumstances could this be an issue with .NET Framework projects?

Reply:
This code is in Common/ so I imagine it is used by the extensions for classic projects?

Reply:
That was what I was thinking.